WCFDataService/Client/System/Data/Services/Client/Materialization/MaterializerEntry.cs

```diff
@@ -275,7 +275,7 @@ public static MaterializerEntry CreateEntryForLoadProperty(EntityDescriptor desc
         /// <returns>The materializer entry</returns>
         public static MaterializerEntry GetEntry(ODataEntry entry)
         {
-            return entry.GetAnnotation<MaterializerEntry>();
+            return entry == null ? null : entry.GetAnnotation<MaterializerEntry>();
```

Comment:
It will be no problem for valid scenario, but for scenarios that would throw exception, with this change, different exception may thrown due to this change. This is kind of breaking change. So to keep it safe, I think it is better to only let it be null when AutoNullPropagation is true. What do you think?